### PR TITLE
Fixes #26254 - Add plugin_present? to template renderer

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: lib/foreman/renderer/scope/macros/base.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def global_setting(name, blank_default = nil)
             (setting.settings_type != "boolean" && setting.value.blank?) ? blank_default : setting.value
           end
 
+          def plugin_present?(name)
+            not Foreman::Plugin.all.select { |plugin| plugin.name == name }.empty?
```
[the remark this answers is not in the document]
Comment (ekohl) [reply lib/foreman/renderer/scope/macros/base.rb]: Should this be `!Foreman::Plugin.all.find { |plugin| plugin.name == name }.nil?` or is there an even more precise way to specify it? :)

**Review by ekohl — CHANGES_REQUESTED** [review]
Could you add a positive test where the plugin is found?

File: lib/foreman/renderer/scope/macros/base.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ares) [reply lib/foreman/renderer/scope/macros/base.rb]: how about `Foreman::Plugin.all.find { |p| p.name == name }.present?`

[the remark this answers is not in the document]
Comment (ares) [reply lib/foreman/renderer/scope/macros/base.rb]: oh the new version also works, please dismiss

File: lib/foreman/renderer/scope/macros/base.rb [hunk 2]
```diff
@@ -21,6 +21,10 @@ def global_setting(name, blank_default = nil)
             (setting.settings_type != "boolean" && setting.value.blank?) ? blank_default : setting.value
           end
 
+          def plugin_present?(name)
+            Foreman::Plugin.all.map(&:name).include?(name)
```
Comment (ares) [inline lib/foreman/renderer/scope/macros/base.rb]: turns out there's already a method for this :-)

```
Foreman::Plugin.find(name).present?
```

I can't think of cleaner way for this

File: lib/foreman/renderer/scope/macros/base.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekohl) [reply lib/foreman/renderer/scope/macros/base.rb]: Your version is probably slightly more efficient but this isn't that performance sensitive anyway.

File: test/unit/foreman/renderer/renderers_shared_tests.rb [hunk 3]
```diff
@@ -39,6 +39,11 @@ module RenderersSharedTests
       assert_equal 'http://foreman.some.host.fqdn', renderer.render(source, @scope)
     end
 
+    test "plugin_present? helper method" do
+      source = OpenStruct.new(content: '<%= plugin_present?("crazy_plugin") %>')
+      assert_equal 'false', renderer.render(source, @scope)
+    end
+
```
Comment (ares) [inline test/unit/foreman/renderer/renderers_shared_tests.rb:53]: this should do for positive test
```
test "plugin_present? finds existing plugin" do
  Foreman::Plugin.register('my_plugin') {}
  source = OpenStruct.new(content: '<%= plugin_present?("my_plugin") %>')
  assert_equal 'true', renderer.render(source, @scope)
end
```

**Review by ares — CHANGES_REQUESTED** [review]
small suggestions that should make @ekohl happy and me able to merge :-)

File: test/unit/foreman/renderer/renderers_shared_tests.rb (hunk 3 above)
Comment (mdellweg) [reply test/unit/foreman/renderer/renderers_shared_tests.rb:53]: Thanks, i'll try that.

Comment (mdellweg) [reply test/unit/foreman/renderer/renderers_shared_tests.rb:53]: It turned out, `Foreman::Plugin.register(:my_plugin) {}` is the way to go. Uff

**Review by ekohl — APPROVED** [review]

**Review by timogoebel — APPROVED** [review]

**Review by ares — APPROVED** [review]

**Review by tbrisker — APPROVED** [review]
Test failures unrelated, thanks @mdellweg !

**Comment by mdellweg** [comment]
Thanks for all the help and quick reviews.

**Pull request merged**